File: Novels/packages/5-mobile-ios/StoryReader/Repository/BookClubsRepository.swift
import Foundation
import Combine

final class BookClubsRepository {
    private let graphQLService = GraphQLService.shared
    
    func getSchedule(groupId: String) -> AnyPublisher<[BookClubScheduleItem], Error> {
        graphQLService.getBookClubSchedule(groupId: groupId)
    }
}

File: Novels/packages/6-mobile-android/StoryReader/app/src/main/java/com/storysphere/storyreader/repository/BookClubsRepository.kt
package com.storysphere.storyreader.repository

import com.storysphere.storyreader.model.BookClubScheduleItem
import com.storysphere.storyreader.network.GraphQLService
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class BookClubsRepository @Inject constructor(
    private val graphQLService: GraphQLService
) {
    suspend fun getSchedule(groupId: String): List<BookClubScheduleItem> =
        graphQLService.getBookClubSchedule(groupId).getOrThrow()
}

